inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp

```diff
@@ -286,7 +286,7 @@ inline void quantizeWeightsBiases(const QuantDesc & quantDesc,
         THROW_IE_EXCEPTION << "Unsupported input dims size for " << wl->name << ", should be > 1, but " << wl->insData[0].lock().get()->getDims().size();
     }
     uint32_t num_rows = isDiagonal ? 1 : wl->outData[0]->getDims()[oIdx];
-    uint32_t num_columns = wl->insData[0].lock().get()->getDims()[iIdx];
+    uint32_t num_columns = isDiagonal ? wl->_weights->size() : wl->insData[0].lock().get()->getDims()[iIdx];
```

Comment:
In diagonal case, number of input columns should be equal to weights size.
If there is mismatch, then most likely it is bug in GNAGraphCompiler::AffinePrimitive

Reply:
In this fix input data for a diagonal layer is reshaped in GNAGraphCompiler::AffinePrimitive to have batch size to be equal to some value from 8 to 1.  wl->insData[0] contains dims before this reshaping, at the same time size of weights created in Pass Manager is equal to input columns number after reshape. So here we need to use weights size which corresponds to actually created weights.

Reply:
thanks for detailed explanation.